webapp/components/_new/generic/ResponsiveImage/ResponsiveImage.vue

```diff
@@ -0,0 +1,87 @@
+<template>
+  <img :data-src="lazySrc" :data-srcset="lazySrcset" :style="style" class="responsive-image lozad" />
+</template>
+
+<script>
+import lozad from 'lozad'
+
+export default {
+  name: 'ResponsiveImage',
+  props: {
+    backgroundColor: {
+      type: String,
+      default: '#efefef',
+    },
+    height: {
+      type: Number,
+      default: null,
+    },
+    lazySrc: {
+      type: String,
+      default: null,
+    },
+    lazySrcset: {
+      type: String,
+      default: null,
+    },
+    width: {
+      type: Number,
+      default: null,
+    },
+  },
+  data() {
+    return {
+      loading: true,
+    }
+  },
+  computed: {
+    aspectRatio() {
+      if (!this.width || !this.height) return null
+      return (this.height / this.width) * 100
+    },
+    style() {
+      const style = { backgroundColor: this.backgroundColor }
+      if (this.width) style.width = `${this.width}px`
+      const applyAspectRatio = this.loading && this.aspectRatio
+      if (applyAspectRatio) {
+        style.height = 0
+        style.paddingTop = `${this.aspectRatio}%`
+      }
+
+      return style
+    },
+  },
+  mounted() {
+    // As soon as the <img> element triggers
+    // the `load` event, the loading state is
+    // set to `false`, which removes the apsect
+    // ratio we've applied earlier.
+    const setLoadingState = () => {
+      this.loading = false
+    }
+    this.$el.addEventListener('load', setLoadingState)
```

Comment:
I don't get it. Why do we have to do this instead of just doing:

```handlebars
<img @load="setLoadingSate">
```

Some examples in this internet look as if that's just fine: https://stackoverflow.com/questions/47051771/does-vue-js-not-trigger-a-load-event-for-img-when-v-if-is-used

Reply:
Or is it because of lozad why we can't use this?